File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.c
int maximumCount(int* nums, int numsSize) 
{
    int neg = 0, nol = 0;
    for (int i = 0; i < numsSize && nums[i] <= 0; ++i)
        if (nums[i] < 0)  
            ++neg;
        else
            ++nol;
    return fmax(neg, numsSize - neg - nol);
}

File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.py
class Solution:
    def maximumCount(self, nums: list[int]) -> int:
        neg: int = 0
        nol: int = 0

        for num in nums:
            if num < 0:
                neg += 1
            elif num > 0:
                break
            else:
                nol += 1
        
        return max(neg, len(nums) - neg - nol)